spec2vec/model_building.py

```diff
@@ -0,0 +1,129 @@
+    sg: int (0,1)
+        For sg = 0 --> CBOW model, for sg = 1 --> skip gram model
+        (see Gensim documentation). Default for Spec2Vec is 0.
+    negative: int
+        from Gensim:  If > 0, negative sampling will be used, the int for
+        negative specifies how many “noise words” should be drawn (usually
+        between 5-20). If set to 0, no negative sampling is used.
+        Default for Spec2Vec is 5.
+    size: int,
+        Dimensions of word vectors Default is 300.
+    window: int,
+        Window size for context words (small for local context, larger for
+        global context). Spec2Vec expects large windwos. Default is 500.
+    min_count: int,
+        Only consider words that occur at least min_count times in the corpus.
+        Default is 1.
+    workers: int,
+        Number of threads to run the training on (should not be more than
+        number of cores/threads. Default is 4.
+    learning_rate_initial: float
+        Set initial learning rate.
+    learning_rate_decay: float
+        After every epoch the learning rate will be lowered by the learning_rate_decay.
+    progress_logger: bool
+        If True, the training progress will be printed every epoch.
+
+    Returns
+    -------
+    word2vec_model
+        Gensim word2vec model.
+    """
+
+    settings = {
+        "sg": 0,
+        "negative": 5,
+        "size": 300,
+        "window": 500,
+        "min_count": 1,
+        "workers": 4,
+        "learning_rate_initial": 0.025,
+        "learning_rate_decay": 0.00025,
+        "progress_logger": True,
+    }
+
+    # Replace default parameters with input
+    for key, value in kwargs.items():
+        if key in settings:
+            print("The value of {} is set from {} (default) to {}".format(key, settings[key], value))
+            settings[key] = value
+
+    num_of_epochs = max(iterations) if isinstance(iterations, list) else iterations
+    alpha, min_alpha = set_learning_rate_decay(settings["learning_rate_initial"],
+                                               settings["learning_rate_decay"], num_of_epochs)
+
+    # Set callbacks
+    callbacks = []
+    if settings["progress_logger"]:
+        training_progress_logger = TrainingProgressLogger(num_of_epochs)
+        callbacks.append(training_progress_logger)
+    if filename:
+        if isinstance(iterations, int):
+            iterations = [iterations]
+        model_saver = ModelSaver(num_of_epochs, iterations, filename)
+        callbacks.append(model_saver)
+
+    # Train word2vec model
+    model = gensim.models.Word2Vec(documents, sg=settings["sg"], negative=settings["negative"],
+                                   size=settings["size"], window=settings["window"],
+                                   min_count=settings["min_count"], workers=settings["workers"],
+                                   iter=num_of_epochs, alpha=alpha, min_alpha=min_alpha,
+                                   seed=321, compute_loss=True, callbacks=callbacks)
```

Comment:
Hardcoding seed is a bit weird to me, as it already defaults to 1 which seem a good seed as well.

Reply:
True, I will remove the default.